tests/neo4j/txrun.py

```diff
@@ -40,3 +38,100 @@ def test_does_not_update_last_bookmark_on_rollback(self):
         tx.rollback()
         bookmarks = self._session.lastBookmarks()
         self.assertEqual(len(bookmarks), 0)
+
+    def test_does_not_update_last_bookmark_on_failure(self):
```

Comment:
Might be a bit out of scope of just migrating tests, but should we also add a tests that describes the behavior for a session that started with a bookmark (either a valid or a made-up one)?

Reply:
It makes sense. I will add it in a next pr.